state.go

```diff
@@ -216,10 +216,10 @@ type currentState struct {
 	roundMessages map[uint64]map[NodeID]*MessageReq
 
 	// Locked signals whether the proposal is locked
-	locked bool
+	locked uint64
```

Comment:
I used the atomic package for changing `locked` variable because it's used in set/getState also(just to be more identical). 
But I agree It may be a possible solution to use sync.Mutex/RWMutex there. Do you see how mutexes make the code better?

Reply:
Basically we won't need to maintain our own mutex so code and tests will be reduced, less code == better code :)